jit-service/src/main/java/org/dcsa/jit/service/OperationsEventService.java

```diff
@@ -29,14 +42,40 @@ public static class OperationsEventFilters {
     String unLocationCode;
     String facilitySMDGCode;
     String operationsEventTypeCode;
-    String eventCreatedDateTime;
+    List<ParsedQueryParameter<OffsetDateTime>> eventCreatedDateTime;
     String sort;
     Integer limit;
     String cursor;
     String apiVersion;
   }
 
-  public List<OperationsEventTO> findAll(OperationsEventFilters operationsEventFilters) {
-    return eventRepository.findAll().stream().map(operationsEventMapper::toTO).toList();
+  public List<OperationsEventTO> findAll(
+      HttpServletRequest request, final OperationsEventFilters requestFilters) {
+
+    Paginator paginator = new Paginator(objectMapper);
+    Cursor c =
```

Comment:
Parsing the cursor should really be in the controller so you dont pass httpservletrequest down to the service.
You should also call paginator.setPageHeaders to set the pagination headers.

Reply:
Why don't we want to have httpservletrequest in service? As far as I can see this is what we've done in multiple places.